pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -197,18 +211,136 @@ func TestReconcileBuildCreationErrors(t *testing.T) {
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			c, _, client := getController(d)
+			c, _, clients := getController(d)
 			if err := c.Reconciler.Reconcile(context.Background(), tc.taskRun); err == nil {
-				t.Error("Expected reconcile to error. Got nil")
+				t.Error("Expected not found error for non exitent task but got nil")
 			}
-			if len(client.Actions()) != 0 {
-				t.Errorf("expected no actions to be created by the reconciler, got %v", client.Actions())
+
+			// build client will fetch build
+			if len(clients.bclient.Actions()) != 1 {
+				t.Errorf("expected no actions to be created by the reconciler, got %v", clients.bclient.Actions())
 			}
 		})
 	}
 
 }
-func getController(d testData) (*controller.Impl, *observer.ObservedLogs, *fakebuildclientset.Clientset) {
+
+func TestReconcileBuildFetchError(t *testing.T) {
+	taskRun := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-taskrun-run-success",
+			Namespace: "foo",
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name:       "test-task",
+				APIVersion: "a1",
+			},
+		},
+	}
+	d := testData{
+		taskruns: []*v1alpha1.TaskRun{
+			taskRun,
+		},
+		tasks: []*v1alpha1.Task{simpleTask},
+	}
+
+	c, _, clients := getController(d)
+
+	reactor := func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+		if action.GetVerb() == "get" && action.GetResource().Resource == "builds" {
+			// handled fetching builds
+			return true, nil, fmt.Errorf("induce failure fetching builds")
+		}
+		return false, nil, nil
+	}
+
+	clients.bclient.PrependReactor("*", "*", reactor)
+
+	if err := c.Reconciler.Reconcile(context.Background(), fmt.Sprintf("%s/%s", taskRun.Namespace, taskRun.Name)); err == nil {
+		t.Fatal("expected error")
+	}
+}
+
+func TestReconcileBuildUpdateStatus(t *testing.T) {
+	taskRun := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-taskrun-run-success",
+			Namespace: "foo",
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name:       "test-task",
+				APIVersion: "a1",
+			},
+		},
+	}
+	d := testData{
+		taskruns: []*v1alpha1.TaskRun{
+			taskRun,
+		},
+		tasks: []*v1alpha1.Task{simpleTask},
+	}
+	buildSt := &duckv1alpha1.Condition{
+		Type: duckv1alpha1.ConditionSucceeded,
+		// build is not completed
+		Status:  corev1.ConditionUnknown,
+		Message: "Running build",
+	}
+
+	c, _, clients := getController(d)
+	build := &buildv1alpha1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      taskRun.Name,
+			Namespace: taskRun.Namespace,
+		},
+		Spec: *simpleTask.Spec.BuildSpec,
+	}
+	build.Status.SetCondition(buildSt)
+
+	_, err := clients.bclient.BuildV1alpha1().Builds(taskRun.Namespace).Create(build)
+	if err != nil {
+		t.Errorf("error creating build : %v", err)
+	}
+
+	if err := c.Reconciler.Reconcile(context.Background(), fmt.Sprintf("%s/%s", taskRun.Namespace, taskRun.Name)); err != nil {
+		t.Fatal("Unexpected error", err)
```

Comment:
should this be fatalf?

Reply:
Yes. If reconcile() has failures then no point continuing the test. I will update the error msg